invenio_rdm_records/services/communities/components.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record communities service components."""
+
+from invenio_communities.communities.records.systemfields.access import (
+    CommunityAccess,
+    VisibilityEnum,
+)
+from invenio_communities.communities.services.components import (
+    CommunityAccessComponent as BaseAccessComponent,
+)
+from invenio_communities.communities.services.components import (
+    CustomFieldsComponent,
+    FeaturedCommunityComponent,
+    OAISetComponent,
+    OwnershipComponent,
+    PIDComponent,
+)
+from invenio_records_resources.services.records.components import (
+    MetadataComponent,
+    RelationsComponent,
+)
+
+from ...proxies import current_rdm_records_service
+from ..errors import InvalidCommunityVisibility
+
+
+class CommunityAccessComponent(BaseAccessComponent):
+    """Community access component."""
+
+    def update(self, identity, data=None, record=None, **kwargs):
+        """Update handler."""
+        self._populate_access_and_validate(identity, data, record, **kwargs)
+        if VisibilityEnum(record.access.visibility) == VisibilityEnum.RESTRICTED:
+            assert CommunityAccess.validate_visibility_level("public")
```

Comment:
I am not sure that I understand this line: is it an extra check to make sure that the keyword `public`, which is used just below, is valid?

Reply:
I've refactored this. The main issue is that the string representation of the visibility `public` comes from the `VisibilityEnum`. At the moment that is coupled in the `CommunityAccess` implementation, but it could potentially change. Or there could be a different more complex implementation of the validation. So it is just to make sure that "it is correct" instead of assuming.